cmd/collectors/zapi/plugins/shelf/shelf.go

```diff
@@ -197,13 +199,23 @@ func (my *Shelf) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 
 					for label, labelDisplay := range my.instanceLabels[attribute].Map() {
 						if value := obj.GetChildContentS(label); value != "" {
-							instance.SetLabel(labelDisplay, value)
+							// This is apply only for the child of the shelf object like psu, fan, etc.
+							// Exclude the instance from metric whose op-status is normal
+							if labelDisplay == "status" && value == "normal" {
+								isInclude = false
+							} else {
+								instance.SetLabel(labelDisplay, value)
+							}
 						}
 					}
 
 					instance.SetLabel("shelf", shelfName)
 					instance.SetLabel("shelf_id", shelfId)
-
+					// Remove instance from metric
+					if !isInclude {
+						data1.RemoveInstance(shelfId + "." + key)
```

Comment:
minor: can you pull out `shelfId + "." + key` into a separate var on the line above and then use the new var on this line and the log line?

Reply:
Done.